src/Domain/CoordinateMaps/TimeDependent/Shape.hpp

```diff
@@ -0,0 +1,162 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "ApparentHorizons/TagsTypeAliases.hpp"
+#include "ApparentHorizons/YlmSpherepack.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/ShapeTransitionFunctions.hpp"
+#include "Utilities/TypeTraits/RemoveReferenceWrapper.hpp"
+#pragma once
```

Comment:
I have changed it to an abstract base class now. Your third idea seems very rigid to me, I am not sure if it would even be possible once we have more than one transition function. The shape map would have to hold all their data but then there would be no way of knowing which data to pass into the transition function